### PR TITLE
docs:Add support rule evaluation via query frontend docs

### LINKED ISSUE #6198: Add documentation for Ruler remote evaluation via Query Frontend
As a follow up of https://github.com/cortexproject/cortex/pull/6151, we need to add relevant doc to document this option.
How it can be enabled, configuration parameters, pros and cons of the rule evaluation via Query Frontend and Ruler.

### PR DESCRIPTION
**Pull request by SungJin1212**
**What this PR does**:
Add docs on configuring Ruler to evaluate rules via Query Frontend and the pros and cons of this feature.
**Which issue(s) this PR fixes**:
Fixes #6198 

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by SungJin1212** [comment]
@yeya24 
Could you take a look?

**Review by friedrichg — APPROVED** [review]
thanks

File: docs/guides/rule-evaluations-via-query-frontend.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+```
+-ruler.frontend-address=query-frontend.svc.cluster.local:9095
+```
+And via yaml:
+```yaml
+ruler:
+  frontend_address: query-frontend.svc.cluster.local:9095
+```
+
+In addition, you can configure gRPC client (Ruler -> Query Frontend) config via args:
```
Comment (yeya24) [inline docs/guides/rule-evaluations-via-query-frontend.md:25]: I think it is unnecessary to list all configuration flags. This adds extra complexity for users who read this doc as I expect that most of the case default values should be good enough. 

Is it possible to use a link to redirect to the frontend client configurations?

File: docs/guides/rule-evaluations-via-query-frontend.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    max_recv_msg_size: 104857600
+    max_send_msg_size: 16777216
+    grpc_compression: ""
+    rate_limit: 0
+    rate_limit_burst: <int> | default = 0
+    backoff_on_ratelimits: <boolean> | default = false
+    backoff_config:
+      min_period: 100ms
+      max_period: 10s
+      max_retries: 10
```
Comment (yeya24) [inline docs/guides/rule-evaluations-via-query-frontend.md:52]: Same as above

File: docs/guides/rule-evaluations-via-query-frontend.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+## Pros and Cons
+
+### Pros
+The rule evaluation performance is improved as we can use Query Frontend features like the vertical query sharding.
```
Comment (yeya24) [inline docs/guides/rule-evaluations-via-query-frontend.md:58]: In terms of performance, I think it is complex and it is not always true to say evaluation performance is improved.

By evaluating rules through Query Frontend, there are more hops in this path. Ruler -> Query Frontend -> Query Scheduler -> Querier -> Ingester/Store Gateway.

Query Frontend/Query Scheduler has a query queue and Querier pulls queries from the queue to execute. If there are not enough queriers, add rule queries to this path could cause query starvation problem and rules cannot get evaluated. I think this is something to call out in this doc.

File: docs/guides/rule-evaluations-via-query-frontend.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+The ruler can use fewer resources as it doesn't need to initialize the querier anymore.
```
Comment (yeya24) [inline docs/guides/rule-evaluations-via-query-frontend.md:59]: This sounds vague to me. You mean in the code Ruler doesn't need to intialize the querier struct? I suggest rewording it to something like:

```
Ruler can use fewer resources as it doesn't need to run query engine to evaluate rules.
```

File: docs/guides/rule-evaluations-via-query-frontend.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+linkTitle: "Rule evaluations via query frontend"
+weight: 10
+slug: rule-evalutions-via-query-frontend
+---
+
+This guide explains how to configure the Ruler to evaluate rules via Query Frontends instead of the Ingesters and the pros and cons of rule evaluation via Query Frontend.
+
+## How to enable
+
+By default, the Ruler queries to the Ingesters for evaluating rules (alerting rules or recording rules). If you have set `-ruler.frontend-address` then the Ruler query to the Query Frontend for evaluation rules.
```
Comment (yeya24) [inline docs/guides/rule-evaluations-via-query-frontend.md:12]: the Ruler queries the Ingesters.
the Ruler queries the Query Frontend.

Ruler queries both Ingesters and Store Gateway depending on the Rule time range.

File: docs/guides/rule-evaluations-via-query-frontend.md (hunk 4 above)
Comment (SungJin1212) [reply docs/guides/rule-evaluations-via-query-frontend.md:59]: Yes, what I mean is that. Let me fix it.

**Comment by SungJin1212** [comment]
@yeya24 
could you review it?

**Comment by SungJin1212** [comment]
@yeya24 
I have renewed the docs, adding `-ruler.query-response-format` contents.

**Comment by yeya24** [comment]
Do we need to update this PR since we have https://github.com/cortexproject/cortex/pull/6345 merged?

**Comment by SungJin1212** [comment]
@yeya24 
I don't think we should update this PR. The `-ruler.query-response-format` is already added.

**Review by yeya24 — APPROVED** [review]
Thanks
